### PR TITLE
Clarify capitalization of Id

### LINKED ISSUE #2456: Clarify why Id is capitalized that way
People keep thinking we've made a mistake, so I think we need to make the explanation harder to miss.

Related: #2455, #2392, #2053

### PR DESCRIPTION
**Pull request by aizatazhar**
Closes #2456

### CODE REVIEW

**Comment by kwalrath** [comment]
To fix the build, we'll need to update /examples/misc/lib/effective_dart/style_good.dart to match.

**Review by munificent — APPROVED** [review]
This looks great, thank you!

**Comment by kwalrath** [comment]
@aizatazhar are you willing to try updating examples/misc/lib/effective_dart/style_good.dart to match your new code, or would you like me to do it?

**Comment by aizatazhar** [comment]
> 
> 
> @aizatazhar are you willing to try updating examples/misc/lib/effective_dart/style_good.dart to match your new code, or would you like me to do it?

Yup I'm willing to give it a shot! Am I supposed to put the good code in examples/misc/lib/effective_dart/style_good.dart and the bad code in put the good code in examples/misc/lib/effective_dart/style_bad.dart?

**Comment by kwalrath** [comment]
> > @aizatazhar are you willing to try updating examples/misc/lib/effective_dart/style_good.dart to match your new code, or would you like me to do it?
> 
> Yup I'm willing to give it a shot!

Great! Thank you!

> Am I supposed to put the good code in examples/misc/lib/effective_dart/style_good.dart and the bad code in examples/misc/lib/effective_dart/style_bad.dart?

Actually, looking at this... It might be easiest just to remove the code-excerpt tag from the .md file. It's going to be a bit difficult to produce .dart code that is easy to excerpt.

File: src/_guides/language/effective-dart/style.md [hunk 1]
```diff
@@ -242,31 +242,35 @@ multiple adjacent acronyms can lead to ambiguous names.
...
-except for two-letter acronyms. (Two-letter *abbreviations* like
-ID and Mr. are still capitalized like words.)
+To avoid this, acronyms and abbreviations are capitalized like regular words.
+
+**Exception:** Two-letter *acronyms* like IO (input/output) are fully 
+capitalized: `IO`. On the other hand, two-letter *abbreviations* like 
+ID (identification) are still capitalized like regular words: `Id`.
 
 {:.good}
 <?code-excerpt "misc/lib/effective_dart/style_good.dart (acronyms and abbreviations)" replace="/,//g"?>
```
Comment (aizatazhar) [inline src/_guides/language/effective-dart/style.md:252]: @kwalrath Sorry, do i just delete this line?

Comment (kwalrath) [reply src/_guides/language/effective-dart/style.md:252]: Yes, please

Comment (aizatazhar) [reply src/_guides/language/effective-dart/style.md]: Done, thank you!

**Comment by kwalrath** [comment]
Thank you, @aizatazhar & @munificent!

**Pull request merged**